plugins/destination/plugin_testing_overwrite_delete_stale.go

```diff
@@ -7,21 +7,26 @@ import (
 
 	"github.com/apache/arrow/go/v12/arrow"
 	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v2/schema"
 	"github.com/cloudquery/plugin-sdk/v2/specs"
 	"github.com/cloudquery/plugin-sdk/v2/testdata"
 	"github.com/cloudquery/plugin-sdk/v2/types"
 	"github.com/google/uuid"
 	"github.com/rs/zerolog"
 )
 
-func (*PluginTestSuite) destinationPluginTestWriteOverwriteDeleteStale(ctx context.Context, p *Plugin, logger zerolog.Logger, spec specs.Destination) error {
+func (*PluginTestSuite) destinationPluginTestWriteOverwriteDeleteStale(ctx context.Context, p *Plugin, logger zerolog.Logger, spec specs.Destination, testSourceOptions testdata.TestSourceOptions) error {
 	spec.WriteMode = specs.WriteModeOverwriteDeleteStale
 	if err := p.Init(ctx, logger, spec); err != nil {
 		return fmt.Errorf("failed to init plugin: %w", err)
 	}
 	tableName := fmt.Sprintf("cq_%s_%d", spec.Name, time.Now().Unix())
-	table := testdata.TestTable(tableName).ToArrowSchema()
-	incTable := testdata.TestTableIncremental(tableName + "_incremental").ToArrowSchema()
+	table := testdata.TestSourceSchema(tableName, testSourceOptions)
+	incrementalMetadata := arrow.MetadataFrom(map[string]string{
+		schema.MetadataTableName:   tableName + "_incremental",
```

Comment:
IMO we mustn't add some custom suffix.
The table name should be declarative.
If we want the suffix we need to add it to the table name instead of adding it to the metadata options).

Reply:
I'm not sure what you're saying here; this isn't new, it's just a refactor of what we had before to use the new method. If you'd like to change the `_incremental` suffix, let's do that separately.

Reply:
I mean, we need to have `tableName := "a_b_c_incremental"` in the future, but yeah, may be left out of scope